copilot-interpreter/src/Copilot/Interpret/Render.hs

```
@@ -69,25 +69,22 @@ step ExecTrace
   where
 
   ppTriggerOutputs :: [Doc]
-  ppTriggerOutputs =
-      catMaybes
-    . fmap ppTriggerOutput
-    . map (fmap head)
-    $ trigs
-
-  ppTriggerOutput :: (String, Maybe [Output]) -> Maybe Doc
-  ppTriggerOutput (_,  Nothing) = Nothing
-  ppTriggerOutput (cs, Just xs) = Just $
+  ppTriggerOutputs = mapMaybe ppTriggerOutput $ trigs
+
+  ppTriggerOutput :: (String, [Maybe [Output]]) -> Maybe Doc
+  ppTriggerOutput (cs, Just xs : _) = Just $
     text cs <> text "," <>
       (foldr (<>) empty . map text . intersperse ",") xs
+  ppTriggerOutput (_,  Nothing : _) = Nothing
+  ppTriggerOutput (_,  []) = Nothing
 
   tails :: Maybe ExecTrace
   tails =
-    if any null (fmap (tail.snd) trigs)
+    if any null (fmap (drop 1.snd) trigs)
```

This changes the behavior if trigs is empty, from a runtime error to a possibly silent error. I'm curious: `tail` throws an exception on non-empty lists, while `drop 1` will return the empty list. Would the sublists in the second component of `trigs` be empty? Is there any protection to make sure those lists are not empty?